app/jobs/regular/process_post.rb

```diff
@@ -60,6 +60,27 @@ def extract_links(post)
       TopicLink.extract_from(post)
       QuotedPost.extract_from(post)
     end
+
+    def auto_tag(post)
+      word_watcher = WordWatcher.new(post.raw)
+
+      old_tags = post.topic.tags.pluck(:name).to_set
+      new_tags = old_tags.dup
+
+      WordWatcher.words_for_action(:tag).each do |word, tags|
+        if word_watcher.word_matches?(word)
+          new_tags += tags.split(",")
+        end
+      end
+
+      if old_tags != new_tags
```

Comment:
Should probably ensure the arrays are `uniq` and `sort`ed?

```suggestion
      if old_tags.uniq.sort != new_tags.uniq.sort
```

Reply:
`old_tags` and `new_tags` are sets, not arrays, and do not require sorting or checking for uniques.

Reply:
My bad, didn't see the `.to_set` 👍